L1Trigger/TrackFindingTracklet/src/MatchCalculator.cc

```diff
@@ -454,19 +454,25 @@ void MatchCalculator::execute() {
       double drphicut = idrphicut * settings_.kphi() * settings_.kr();
       double drcut = idrcut * settings_.krprojshiftdisk();
 
-      if (settings_.writeMonitorData("Residuals")) {
-        double pt = 0.01 * settings_.c() * settings_.bfield() / std::abs(tracklet->rinv());
+      bool match, imatch;
+      if(std::abs(dphi) < 0.25 && std::abs(dphiapprox) < 0.25){ //Changed the Asserts into if statements
+        if (settings_.writeMonitorData("Residuals")) {
+          double pt = 0.01 * settings_.c() * settings_.bfield() / std::abs(tracklet->rinv());
 
-        globals_->ofstream("diskresiduals.txt")
-            << disk << " " << stub->isPSmodule() << " " << tracklet->layer() << " " << abs(tracklet->disk()) << " "
-            << pt << " " << ideltaphi * settings_.kphi() * stub->r() << " " << drphiapprox << " " << drphicut << " "
-            << ideltar * settings_.krprojshiftdisk() << " " << deltar << " " << drcut << " " << endl;
-      }
-
-      bool match = (std::abs(drphi) < drphicut) && (std::abs(deltar) < drcut);
+          globals_->ofstream("diskresiduals.txt")
+              << disk << " " << stub->isPSmodule() << " " << tracklet->layer() << " " << abs(tracklet->disk()) << " "
+              << pt << " " << ideltaphi * settings_.kphi() * stub->r() << " " << drphiapprox << " " << drphicut << " "
+              << ideltar * settings_.krprojshiftdisk() << " " << deltar << " " << drcut << " " << endl;
+        }
 
-      bool imatch = (std::abs(ideltaphi * irstub) < idrphicut) && (std::abs(ideltar) < idrcut);
+        match = (std::abs(drphi) < drphicut) && (std::abs(deltar) < drcut);
 
+        imatch = (std::abs(ideltaphi * irstub) < idrphicut) && (std::abs(ideltar) < idrcut);
```

Comment:
Compiler says imatch may be undefined at L482. I think it is right!

Reply:
Yes, Sorry about that.
I Fixed it,